sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,138 @@
-import {
-  AppBar,
-  AppBarProps,
-  Backdrop,
-  IconButton,
-  paperClasses,
-  Popover,
-  PopoverProps,
-  Typography
-} from '@mui/material';
-import CloseRounded from '@mui/icons-material/CloseRounded';
+import { AppBarProps, paperClasses, Popover, PopoverProps, useTheme } from '@mui/material';
 import ChatGPT, { ChatGPTProps } from './ChatGPT.tsx';
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import MinimizedBar from '@craftercms/studio-ui/components/MinimizedBar';
+import DialogHeader from '@craftercms/studio-ui/components/DialogHeader/DialogHeader';
+import AlertDialog from '@craftercms/studio-ui/components/AlertDialog';
+import PrimaryButton from '@craftercms/studio-ui/components/PrimaryButton/PrimaryButton';
+import SecondaryButton from '@craftercms/studio-ui/components/SecondaryButton/SecondaryButton';
+import { useSelector } from 'react-redux';
+import { GlobalState } from '@craftercms/studio-ui';
 
 export interface ChatGPTPopoverProps extends PopoverProps {
   appBarTitle?: string;
   width?: number | string;
   height?: number | string;
   appBarProps?: AppBarProps;
-  chatGPTProps: ChatGPTProps;
+  chatGPTProps?: ChatGPTProps;
+  isMinimized?: boolean;
+  onMinimize?: () => void;
+  onMaximize?: () => void;
 }
 
-function ChatGPTPopover(props: ChatGPTPopoverProps) {
+function ChatGPTPopover(props: Readonly<ChatGPTPopoverProps>) {
+  const theme = useTheme();
+  const dialogsState = useSelector((state: GlobalState) => state.dialogs);
   const {
     open,
     onClose,
-    appBarProps,
     chatGPTProps,
+    isMinimized = false,
+    onMinimize,
+    onMaximize,
     appBarTitle = 'AI Assistant',
     width = 450,
     height = 500,
     ...popoverProps
   } = props;
+
+  const chatGptRef = useRef<{ hasConversation: () => boolean }>(null);
+  const [openAlertDialog, setOpenAlertDialog] = useState(false);
+
+  // In case the Edit form is opened, make sure the chat is minimized
+  useEffect(() => {
+    if (open && dialogsState.edit.open && !dialogsState.edit.isMinimized) {
+      onMinimize?.();
+    }
+  }, [dialogsState, onMinimize, open]);
```

Comment:
Please remove this. If you have questions about the previous message or how to solve a specific problem, ping me on slack.

Reply:
Rolled back this for now